apps/files/decorators.py

```diff
@@ -18,21 +18,22 @@
 
 
 def allowed(request, file):
-    allowed = acl.check_addons_reviewer(request)
-    if not allowed:
-        try:
-            addon = file.version.addon
-        except ObjectDoesNotExist:
-            raise http.Http404
-
+    try:
+        addon = file.version.addon
+    except ObjectDoesNotExist:
+        raise http.Http404
+
+    if addon.is_listed and acl.check_addons_reviewer(request):
```

Comment:
Not a fan of the multiple if clauses and multiple `acl.` checks in that function. It feels there is some redundancy and could be refactored.